### PR TITLE
Unicode error in dmnotify client

### PR DESCRIPTION
**Pull request by TheDoubleK**
This should hopefully fix the error you can see at the bottom of the log here: https://ci.beta.digitalmarketplace.service.gov.uk/job/generate-upload-and-notify-counterpart-signature-pages-production/25/consoleFull

### CODE REVIEW

**Comment by TheDoubleK** [comment]
Sorry, yep the `six` was something I started on and then decided against.  Removed now.

**Comment by TheDoubleK** [comment]
@allait adding unicode to test data threw up another issue - inconsistent reference generation between Py2 ad Py3.  I've fixed it by using explicit utf-8 encoding rather than `str` function.  Does this look reasonable?

File: tests/email/test_dm_notify.py [hunk 1]
```diff
@@ -59,10 +59,10 @@ def notify_send_email():
         },
         'uri': 'https://api.notifications.service.gov.uk/v2/notifications/ceb50d92-100d-4b8b-b559-14fa3b091cd',
         'template': {
-             'id': 'ceb50d92-100d-4b8b-b559-14fa3b091cda',
-             'version': 1,
-             'uri': 'https://api.notificaitons.service.gov.uk/service/'
-             'your_service_id/templates/bfb50d92-100d-4b8b-b559-14fa3b091cda'
+            'id': 'ceb50d92-100d-4b8b-b559-14fa3b091cda',
+            'version': 1,
+            'uri': 'https://api.notificaitons.service.gov.uk/service/'
```
Comment (pcraig3) [inline tests/email/test_dm_notify.py]: notificaitons :/

This is purely a test string, right?

Comment (TheDoubleK) [inline tests/email/test_dm_notify.py]: All I've changed is the indentation here 😛

Comment (TheDoubleK) [inline tests/email/test_dm_notify.py]: But fixed now.

Comment (pcraig3) [inline tests/email/test_dm_notify.py]: Yes!

File: dmutils/email/dm_notify.py [hunk 2]
```diff
@@ -62,8 +62,10 @@ def get_reference(email_address, template_id, personalisation=None, **kwargs):
         :param kwargs: Extra data passed to the reference eg. {'notes': 'Manual resend'}
         :return: Hashed string 'reference' to be passed to client.send_email_notification or self.send_email
         """
-        personalisation_string = ','.join(list(map(str, personalisation.values()))) if personalisation else ''
-        details_string = '|'.join([email_address, template_id, personalisation_string])
+        personalisation_string = u','.join(
+            list(map(lambda x: x.format('utf-8'), personalisation.values()))
```
Comment (allait) [inline dmutils/email/dm_notify.py]: Hey, sorry, I totally missed this on first read, but I think this should be `.encode('utf-8')`

**Pull request merged**